lib/imagefs.coffee

```diff
@@ -23,16 +23,71 @@ _ = require('lodash')
 filedisk = require('file-disk')
 fs = Promise.promisifyAll(require('fs'))
 replaceStream = require('replacestream')
+ext2fs = require('ext2fs')
+
 driver = require('./driver')
+utils = require('./utils')
 
 checkImageType = (image) ->
 	if not (_.isString(image) or image instanceof filedisk.Disk)
 		throw new Error('image must be a String (file path) or a filedisk.Disk instance')
 
+composeDisposers = (outerDisposer, createInnerDisposer) ->
+	Promise.resolve(outerDisposer)
+	.then (outerDisposer) ->
+		outerDisposer._promise
+		.then (outerResult) ->
+			Promise.resolve(createInnerDisposer(outerResult))
+			.then (innerDisposer) ->
+				innerDisposer._promise
+				.then (innerResult) ->
+					Promise.resolve(innerResult)
+					.disposer (innerResult) ->
+						Promise.resolve(innerDisposer._data(innerResult))
+						.then ->
```

Comment:
> I guess you can circumvent this by explicitly returning null – I'd also recommend adding a comment about why this is supposed to be a noop
I already changed this with an explicit return. I'll add a comment too.

> I have a hard time making sense of this promisery, but I get the impression this doesn't need to be this complicated – maybe @jviotti has suggestions?
The idea is to use a disposer in a disposer. I don't think this can be made much simpler. See https://github.com/petkaantonov/bluebird/issues/1379 .